pkg/storaged/details.js

```diff
@@ -48,7 +48,24 @@
                 return client.mdraids[path].Start({ "start-degraded": { t: 'b', v: true } });
             },
             mdraid_stop: function mdraid_stop(path) {
-                return client.mdraids[path].Stop({});
+                var mdraid = client.mdraids[path];
+                var block = client.mdraids_block[path];
+                var usage = block? utils.get_usage_alerts(client, block.path) : [ ];
+
+                if (usage.length === 0)
+                    return mdraid.Stop({});
+
+                dialog.open({ Title: cockpit.format(_("Please confirm stopping of $0"),
+                                                    utils.mdraid_name(mdraid)),
```

Comment:
> Shouldn't this point out that the array is currently being used?

It does.  The dialog will show the warning based on the `Alerts: usage` option.

> And if the user then confirms anyway, wouldn't she get the same "unexpected error" from the original issue again? Might be safer to just say that it's busy (show usage somewhere if that's something a human can interpret?) and ask the user to unmount it before stopping?

That's a good question, but it applies to all of the storage page.  Right now, we take the approach that we don't know everything and let the user do things that will probably fail.
fail.

I would argue that the knowledge whether something will fail with certainty should be in UDisks2.  Otherwise we go too much storage expertise in Cockpit that would be better shared between the clients of UDisks2.